### PR TITLE
Cloud Pub/Sub Notifications support

### PR DESCRIPTION
**Pull request by akm**
From #47 

This PR have `blocks-gcs-proxy` support [Cloud Pub/Sub Notifications for Google Cloud Storage](https://cloud.google.com/storage/docs/pubsub-notifications).
Add [an example](https://github.com/groovenauts/blocks-gcs-proxy/blob/features/cloud_pubsub_notification_support/examples/pubsub_notifications/README.md) to try this feature.

See the [document](https://github.com/groovenauts/blocks-gcs-proxy/blob/features/cloud_pubsub_notification_support/doc/pubsub_notification.md) for more detail.

### CODE REVIEW

**Review by totty — APPROVED** [review]

**Review by nagachika — APPROVED** [review]

**Review by taigou — APPROVED** [review]

File: examples/pubsub_notifications/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+$ export PIPELINE=pipeline01
+$ export PROJECT=your-gcp-project
+$ export TOPIC="projects/${PROJECT}/topics/${PIPELINE}-job-topic"
+$ export BUCKET="your-bucket"
+$ gsutil notification create -t $TOPIC -f json -e OBJECT_FINALIZE gs://$BUCKET
+$ gsutil cp [path/to/localFile] $BUCKET/path/to/remoteFile
+```
+
+gsutil notification create -t projects/scenic-doodad-617/topics/akm-pipeline01-job-topic                -f json gs://blocks-gcs-proxy-test -e OBJECT_FINALIZE
+gsutil notification create -t projects/scenic-doodad-617/topics/akm-test-gcs-pubsub-notifications-topic -f json gs://akm-test
```
Comment (minimum2scp) [inline examples/pubsub_notifications/README.md]: Are last 2 lines necessary?

File: examples/pubsub_notifications/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# [config] VERSION_SCRIPT: 'grep VERSION ../../version.go | cut -f2 -d\"'
+# [config] GIT_TAG_PREFIX: 'examples/pubsub_notifications/'
+
+FROM buildpack-deps:jessie-curl
+
+ENV APP_HOME /usr/app/batch_type_example
+COPY . $APP_HOME
+WORKDIR $APP_HOME
+
+ENV BLOCKS_GCS_PROXY_VERSION 0.6.0-alpha1
```
Comment (guemon) [inline examples/pubsub_notifications/Dockerfile]: `0.6.0-alpha1` -> `0.6.0`

File: examples/pubsub_notifications/README.md (hunk 1 above)
Comment (akm) [reply examples/pubsub_notifications/README.md]: @minimum2scp  oops! I removed them. Thank you!

File: examples/pubsub_notifications/Dockerfile (hunk 2 above)
Comment (akm) [reply examples/pubsub_notifications/Dockerfile]: @guemon Thanks! I fixed it.

**Review by minimum2scp — APPROVED** [review]

**Review by guemon — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -28,7 +28,9 @@ endif
 
 build:
 	mkdir -p ${PKGDIR}
-	gox -output="${PKGDIR}/{{.Dir}}_${OS}_${ARCH}" -os="${OS}" -arch="${ARCH}"
+	for x in "$(OS_LIST)" ; do \
+		gox -output="${PKGDIR}/{{.Dir}}_{{.OS}}_{{.Arch}}" -os="$$x" -arch="${ARCH}" ; \
+	done
```
Comment (ko1nksm) [inline Makefile]: @akm 

```for x in "$(OS_LIST)" ; do \``` expand to  ```for x in "linux darwin" ; do \```
therefore always loops one time.

gox's -os parameter accepts multiple values saparated by space.

> gox --help
>
> -os=""              Space-separated list of operating systems to build for

so, this code will work well, but meaningless "for".

Comment (akm) [reply Makefile]: @ko1nksm Thanks, I removed `for` statement.

**Review by ko1nksm — APPROVED** [review]

**Review by hokuts — APPROVED** [review]

**Pull request merged**